src/java_bytecode/java_bytecode_convert_method.cpp

```diff
@@ -248,6 +248,49 @@ const exprt java_bytecode_convert_methodt::variable(
   }
 }
 
+// In order to construct the method type, we can either use signature or
+// descriptor. Since only signature contains the generics info, we want to
+// use signature whenever possible. We revert to using descriptor if (1) the
+// signature does not exist, (2) an unsupported generics are present or
+// (3) in the special case when the number of parameters in the descriptor does
+// not match the number of parameters in the signature - e.g. for certain
+// types of inner classes and enums (see unit tests for examples).
+typet member_type_lazy(const std::string &descriptor,
+                       const optionalt<std::string> &signature,
+                       const std::string &class_name,
+                       message_handlert &message_handler)
+{
+  messaget message;
+  message.set_message_handler(message_handler);
```

Comment:
Can't you do `messaget message(message_handler)` here? (I've currently got an out of date branch checked out so I might be looking at an old constructor here.

Reply:
You're right, that's the correct constructor, I will correct this. In order to use it, I cannot make the `const` though.